kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/encryption/EnvelopeEncryptionFilterIT.java

```diff
@@ -338,28 +257,19 @@ void shouldGenerateOneDek(KafkaCluster cluster) throws Exception {
                     .isEqualTo(message);
         }
 
-        // We expect that only one DEK was generated
-        assertThat(kms.numDeksGenerated()).isEqualTo(1);
+        assertThat(testKmsFacade.getKms())
+                .asInstanceOf(InstanceOfAssertFactories.type(InMemoryKms.class))
+                .extracting(InMemoryKms::numDeksGenerated).isEqualTo(1);
 
-        IntegrationTestingKmsService.delete(kmsId.toString());
     }
 
-    /*
-     *
-     * further IT ideas:
-     * fetching from > 1 topics (mixed encryption/plain case)
-     * exploratory test examining what the client will see/do when decryption fails - looking to verify
-     * - behaviour is reasonable
-     * - the user has a chance to understand what's wrong.
-     * behaviour when KMS throws a KmsException a) during produce, b) during fetch
-     */
-    private FilterDefinition encryptionFilterDefinition(UUID kmsId) {
-        return new FilterDefinitionBuilder("EnvelopeEncryption")
-                .withConfig("kms", IntegrationTestingKmsService.class.getSimpleName())
-                .withConfig("kmsConfig", Map.of("name", kmsId.toString()))
+    private FilterDefinition getEncryptionFilterDefinition(TestKmsFacade<?, ?, ?> testKmsFacade) {
```

Comment:
personal preference nit:

```suggestion
    private FilterDefinition buildEncryptionFilterDefinition(TestKmsFacade<?, ?, ?> testKmsFacade) {
```

get implies accessing something that exists already and thus shared but `build` implies that its being constructed for the caller.

Reply:
done.